src/screens/Home/surveyList.js

```diff
@@ -1,53 +1,102 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useContext, useEffect, useRef, useState } from 'react';
 
 import Slider from 'react-slick';
 
-import testSurvey from 'images/auth_background.png';
+import { AuthContext } from 'contexts/auth';
+import { BackgroundContext } from 'contexts/background';
 import slickSettings from 'screens/Home/slickSettings';
+import Api from 'utils/Api';
+
+const SurveyList = ({ surveyResponse }) => {
+  const surveyData = surveyResponse.data;
+  const surveyMeta = surveyResponse.meta;
+  const total_pages = surveyMeta.pages;
 
-const SurveyList = () => {
   const sliderRef = useRef(null);
 
+  const [currentPage, setCurrentPage] = useState(surveyMeta.page);
+  const [list, setList] = useState(surveyData);
+  const [currentBackground, setCurrentBackground] = useState(
+    list[0].attributes.cover_image_url
+  );
+
+  const { dispatch } = useContext(BackgroundContext);
+  const authContext = useContext(AuthContext);
+
+  const sliderEvents = {
+    beforeChange: (current, next) => beforeSlideChange(next),
+    afterChange: (current) => afterSlideChange(current),
+  };
+  const slickConfig = { ...slickSettings, ...sliderEvents };
+
   useEffect(() => {
     window.addEventListener('wheel', (e) => {
       slideList(e.wheelDelta);
-    })
-  }, []);
+    });
+
+    dispatch({
+      type: 'SURVEY',
+      payload: currentBackground,
+    });
+  }, [currentBackground, dispatch]);
+
+  const beforeSlideChange = (next) => {
+    setCurrentBackground(list[next].attributes.cover_image_url);
+  };
+
+  const afterSlideChange = () => {
+    if (currentPage < total_pages) {
```

Comment:
@rafayet-monon When we set `speed: 500`, it seems we have a little bit strange for me.
For example, I'm on the 3rd slide then I scroll the screen, it skips the slide to the 3(+-)2 slide.
It should skip to 3(+-)1 slide (only one step)